packages/ra-core/src/util/getFieldLabelTranslationArgs.ts

```diff
@@ -37,6 +45,10 @@ export default (options?: Args): TranslationArguments => {
         ['underscore', 'humanize']
     );
 
+    if (labelFromSourceContext) {
+        return [labelFromSourceContext, { _: defaultLabel }];
+    }
+
```

Comment:
Because of early return on L37, I'm afraid this won't work.

Reply:
It does, if you didn't provide your own label